src/skvhttpclient/src/skvhttp/mpack/MPackSerialization.h

```diff
@@ -554,6 +575,35 @@ class MPackWriter {
     MPackWriter() {
         mpack_writer_init_growable(&_writer, &_data, &_size);
     }
+
+    ~MPackWriter() {
+        // memory is copied to data only when writer is destroyed
+        mpack_writer_destroy(&_writer);
+        MPACK_FREE(_data);
+    }
+
+
+    MPackWriter& operator=(const MPackWriter& o) = delete;
+
+    MPackWriter(const MPackWriter& o) = delete;
+
+    MPackWriter(MPackWriter&& o) {
+        _data = o._data;
+        _size = o._size;
+        o._data = NULL;
+        o._size = 0;
+        _writer = std::move(o._writer);
+    }
+
+    MPackWriter& operator=(MPackWriter&& o) {
+        _data = o._data;
+        _size = o._size;
+        o._data = NULL;
+        o._size = 0;
+        _writer = std::move(o._writer);
```

Comment:
you need to call mpack_writer_destroy as well as MPACK_FREE here (like in the dtor)